tests/unit/bokeh/command/subcommands/test_serve.py

```diff
@@ -471,6 +480,33 @@ def check_error(args):
         pytest.fail(f"command {cmd} unexpected successful")
     return out
 
+def test_unix_socket_with_port() -> None:
+    unix_socket = "test.sock"
+    out = check_error(["--unix-socket", unix_socket, "--port", "5000"])
+    expected = "--port arg is not supported with a unix socket\n"
+    assert expected == out
+
+def test_unix_socket_with_invalid_args() -> None:
+    invalid_args = ["address", "allow-websocket-origin"]
+    for arg in invalid_args:
+        unix_socket = "test.sock"
+        out = check_error(["--unix-socket", unix_socket, f"--{arg}", "value"])
+        expected = "['address', 'allow_websocket_origin', 'port'] args are not supported with a unix socket\n"
+        assert expected == out
+
+def test_unix_socket() -> None:
+    sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+    file_name = join(HERE, "test.socket")
+    sock.bind(file_name)
+    with run_bokeh_serve(["--unix-socket", file_name, "--glob", APPS]) as (p, nbsr):
+        # The server is not ready is binds to the unix socket
+        # very quickly, having some sleep helps
```

Comment:
If there's not a better option here then we should probably also add the `flaky` decorator to the test. cc @mattpap for any thoughts though

Reply:
Added the flaky decorator for now - will update if there are any other thoughts